### PR TITLE
add maxcompute project resource

### PR DESCRIPTION
**Pull request by superMichelle7**
add maxcompute project resource, currently support project create only。

### CODE REVIEW

File: alicloud/provider.go [hunk 1]
```diff
@@ -567,6 +568,7 @@ func providerConfigure(d *schema.ResourceData) (interface{}, error) {
 		config.CasEndpoint = strings.TrimSpace(endpoints["cas"].(string))
 		config.MarketEndpoint = strings.TrimSpace(endpoints["market"].(string))
 		config.AdbEndpoint = strings.TrimSpace(endpoints["adb"].(string))
+		config.MaxComputeEndpoint = strings.TrimSpace(endpoints["maxcompute"].(string))
```
Comment (Lexsss) [inline alicloud/provider.go]: add the endpoint to line 731, 1021 and 1069

File: alicloud/connectivity/client.go [hunk 2]
```diff
@@ -1658,3 +1660,35 @@ func (client *AliyunClient) WithAdbClient(do func(*adb.Client) (interface{}, err
...
+		if endpoint != "" {
+			endpoints.AddEndpointMapping(client.config.RegionId, string(MAXCOMPUTECode), endpoint)
+		}
+		maxcomputeconn, err := maxcompute.NewClientWithOptions(client.config.RegionId, client.getSdkConfig(), client.config.getAuthCredential(false))
+		if err != nil {
+			return nil, fmt.Errorf("unable to initialize the MaxCompute client: %#v", err)
+		}
+
+		maxcomputeconn.AppendUserAgent(Terraform, terraformVersion)
+		client.maxcomputeconn = maxcomputeconn
```
Comment (Lexsss) [inline alicloud/connectivity/client.go]: why no `providerVersion ` and `client.config` append?

File: alicloud/resource_alicloud_maxcompute_project.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+	d.SetId(request.ProjectName)
+
+	return resourceAliyunMaxComputeProjectRead(d, meta)
+}
+
+func resourceAliyunMaxComputeProjectRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+
+	request := maxcompute.CreateGetProjectRequest()
```
Comment (Lexsss) [inline alicloud/resource_alicloud_maxcompute_project.go]: use service method please

File: alicloud/resource_alicloud_maxcompute_project.go [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(3, 27),
+			},
+
+			"specification_type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{"OdpsStandard"}, false),
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_maxcompute_project.go]: If specification_type only support one valid value, please set it to optional and add a default for it.

File: alicloud/resource_alicloud_maxcompute_project.go [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+			},
+
+			"order_type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{"PayAsYouGo"}, false),
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_maxcompute_project.go]: still applies

File: alicloud/resource_alicloud_maxcompute_project.go [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+	return resourceAliyunMaxComputeProjectRead(d, meta)
+}
+
+func resourceAliyunMaxComputeProjectRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	maxcomputeService := MaxComputeService{client}
+	response, err := maxcomputeService.DescribeMaxComputeProject(d.Id())
+	if err != nil {
+		return WrapError(err)
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_maxcompute_project.go]: There needs to checking whether the project exist. If not, there needs to set id to "" and return nil

File: alicloud/resource_alicloud_maxcompute_project_test.go [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+		Providers:     testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccMaxcomputeConfigBasic,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheck(map[string]string{
+						"name":               name,
+						"specification_type": "OdpsStandard",
+						"order_type":         "PayAsYouGo",
+					}),
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_maxcompute_project_test.go]: please add import testcase

File: website/docs/r/maxcompute_project.html.markdown [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+sidebar_current: "docs-alicloud-resource-maxcompute-project"
+description: |-
+  Provides a Alicloud maxcompute project resource.
+---
+
+# alicloud\_maxcompute\_project
+
+The project is the basic unit of operation in maxcompute. It is similar to the concept of Database or Schema in traditional databases, and sets the boundary for maxcompute multi-user isolation and access control. [Refer to details](https://www.alibabacloud.com/help/doc-detail/27818.html).
+
+->**NOTE:** Available in 1.74.0+.
```
Comment (xiaozhu36) [inline website/docs/r/maxcompute_project.html.markdown]: The next version is 1.75.0+

Comment (superMichelle7) [reply website/docs/r/maxcompute_project.html.markdown]: done

File: alicloud/resource_alicloud_maxcompute_project_test.go (hunk 7 above)
Comment (superMichelle7) [reply alicloud/resource_alicloud_maxcompute_project_test.go]: done

**Pull request merged**